### DISCUSSION TITLE
KPMODE=0 or KPMODE=1 fails with anphon RTA calculation

### DISCUSSION THREAD

**Discussion by irrotational** [ttadano/alamode#216] [Q&A] [answered]
I'm trying to get a phonon dispersion plot along high symmetry directions, with linewidths superimposed.

I've followed the Silicon tutorial with no problems, and the calculation works just fine if I use KPMODE=2, as per the tutorial. However, if I change the &kpoint mode to KPMODE=1 to get kpoints along a high symmetry path, it fails with:

```
*** Process received signal ***
Signal: Segmentation fault (11)
Signal code: Address not mapped (1)
Failing at address: 0x10
[ 0] /lib/x86_64-linux-gnu/libc.so.6(+0x42520)[0x7f6c8629b520]
[ 1] anphon(_ZNK7PHON_NS7Isotope27calc_isotope_selfenergy_allEv+0x49)[0x558a65f4847d]
[ 2] anphon(_ZNK7PHON_NS4PHON11execute_RTAEv+0x28a)[0x558a65fb194c]
[ 3] anphon(_ZN7PHON_NS4PHONC1EiPPcP19ompi_communicator_t+0x3b5)[0x558a65fb052f]
[ 4] anphon(main+0x4e)[0x558a65f6fe41]
[ 5] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90)[0x7f6c86282d90]
[ 6] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80)[0x7f6c86282e40]
[ 7] anphon(_start+0x25)[0x558a65e38bb5]
*** End of error message ***
```

It fails with a near-identical message if I manually specify kpoints with KPMODE=0. My anphon RTA input file is:

```
&general
  PREFIX = si222
  MODE = RTA
  FCSXML = si222_cubic.xml
  NKD = 1; KD = Si
  MASS = 28.0855
/

&cell
  10.203
  0.0 0.5 0.5
  0.5 0.0 0.5
  0.5 0.5 0.0
/

&kpoint
  1  # KPMODE = 1: line mode
  G 0.0 0.0 0.0 X 0.5 0.5 0.0 51
  X 0.5 0.5 1.0 G 0.0 0.0 0.0 51
  G 0.0 0.0 0.0 L 0.5 0.5 0.5 51
/
```

Since KPMODE=2 works OK, I suppose I could manually interpolate the resulting grid onto high symmetry directions, but it seems a shame not to be able to just use KPMODE=1 for a high symmetry path.

**Comment by ttadano** [comment, marked as answer]
Thank you for the report. `MODE = RTA` is not supposed to be used with `KPMODE = 0, 1`. 

We will change the code so that it stops when `MODE =RTA` is used with `KPMODE = 0, 1`.

**Reply by irrotational** [reply]
Thanks so much for the quick response.

Is there any other way to get the phonon lifetimes along high symmetry directions? Or will I just have to interpolate it myself using the data in the `.result` file?
